packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_prop/string_prop.rs

```diff
@@ -72,20 +123,39 @@ impl PropUpdater<String, RequiredData> for StringProp {
         self.default_value.clone()
     }
 
-    fn return_data_queries(&self) -> Vec<Option<DataQuery>> {
+    fn return_data_queries(&self) -> Vec<DataQuery> {
         RequiredDataQueries {
-            strings: Some(self.data_query.clone()),
+            independent_state: DataQuery::State,
```

Comment:
I hope we can figure out another way to do this...

Reply:
We need a place to store changes when there aren't dependencies. One way is with the "state props" that get snuck into children/attributes when there are no components (if `always_return_value` is `true`). If we don't have that turned on (because we want to treat the case with no components differently), then we have to create another "state prop" to store changes.